client/src/components/EditPlace.js

```diff
@@ -1,43 +1,99 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { history } from '../history';
+import {
+  getPlace,
+  getLists,
+  updatePlace,
+  addPlaceToList,
+  removePlaceFromList,
+} from '../redux/actions';
 import { InputText } from 'primereact/inputtext';
 import { InputTextarea } from 'primereact/inputtextarea';
 import { Button } from 'primereact/button';
 import { Dropdown } from 'primereact/dropdown';
 import { Rating } from 'primereact/rating';
-import { place as fakePlace } from '../fakeData/place';
-import { database } from '../fakeData/database';
 import '../styles/NewPlace.css';
 
 export class EditPlace extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { list: null, rating: fakePlace.rating };
+    this.placeId = this.props.match.params.id;
+    this.state = {
+      fetchedPlace: null,
+      name: '',
+      rating: null,
+      description: '',
+      location: '',
+      selectedList: null,
+    };
   }
 
-  handleClickCancel = () => {
-    this.props.history.push('/');
+  componentDidMount() {
+    this.props.getPlace(this.placeId);
+    this.props.getLists();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.place !== prevProps.place) this.setFetchedPlace();
```

Comment:
Nitpick, don't use this style. Use curly braces. This will break during minification.